### PR TITLE
LRO objects and fake LRO objects bind with the corresponding resource's name

### PR DESCRIPTION
**Pull request by ArcturusZhang**
Fixes https://github.com/Azure/autorest.csharp/issues/1719

# Description

# Checklist

To ensure a quick review and merge, please ensure:
- [ ] The PR has a understandable title and description explaining the _why_ and _what_.
- [ ] The PR is opened in draft if not ready for review yet.
   - If opened in draft, please allocate sufficient time (24 hours) after moving out of draft for review
- [ ] The branch is recent enough to not have merge conflicts upon creation.

# Ready to Land?
- [ ] Build is completely green
   - Submissions with test failures require tracking issue and approval of a CODEOWNER
- [ ] At least one +1 review by a CODEOWNER
- [ ] All -1 reviews are confirmed resolved by the reviewer 
   - Override/Marking reviews stale must be discussed with CODEOWNERS first

### CODE REVIEW

File: src/AutoRest.CSharp/MgmtTest/Generation/MgmtBaseTestWriter.cs [hunk 1]
```diff
@@ -659,31 +659,31 @@ public List<string> WriteOperationParameters(IEnumerable<Parameter> methodParame
...
-                var returnType = new CSharpType(typeof(IReadOnlyList<>), actualItemType);
-                return GetResponseType(returnType, async);
-            }
-            else
-            {
-                var returnType = WrapResourceDataType(clientOperation.ReturnType, clientOperation.First())?.Type ?? clientOperation.ReturnType;
-                return GetResponseType(returnType, async);
-            }
-        }
+        //public CSharpType GenResponseType(MgmtClientOperation clientOperation, bool async, string? methodName = null)
```
Comment (m-nash) [inline src/AutoRest.CSharp/MgmtTest/Generation/MgmtBaseTestWriter.cs:662]: Is this supposed to be commented out?

**Review by m-nash — CHANGES_REQUESTED** [review]

File: src/AutoRest.CSharp/MgmtTest/Generation/MgmtBaseTestWriter.cs (hunk 1 above)
Comment (ArcturusZhang) [reply src/AutoRest.CSharp/MgmtTest/Generation/MgmtBaseTestWriter.cs]: I removed this piece of code.

File: test/AutoRest.Shared.Tests/AutoRest - Backup.Shared.Tests.csproj [hunk 2]
```diff
@@ -0,0 +1,23 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
```
Comment (m-nash) [inline test/AutoRest.Shared.Tests/AutoRest - Backup.Shared.Tests.csproj:1]: remove?

File: test/AutoRest.Shared.Tests/AutoRest.Shared.Tests.csproj [hunk 3]
```diff
@@ -1,10 +1,9 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk">
 
   <PropertyGroup>
-    <TargetFramework>netcoreapp3.1</TargetFramework>
+    <TargetFrameworks>netcoreapp3.1;net6.0</TargetFrameworks>
```
Comment (m-nash) [inline test/AutoRest.Shared.Tests/AutoRest.Shared.Tests.csproj:4]: @ArthurMa1978 PR just removed this are we sure we want to add it back?

**Review by m-nash — CHANGES_REQUESTED** [review]

File: test/AutoRest.Shared.Tests/AutoRest - Backup.Shared.Tests.csproj (hunk 2 above)
Comment (ArcturusZhang) [reply test/AutoRest.Shared.Tests/AutoRest - Backup.Shared.Tests.csproj:1]: Sorry about this... VS introduces this

File: test/AutoRest.Shared.Tests/AutoRest.Shared.Tests.csproj (hunk 3 above)
Comment (ArcturusZhang) [reply test/AutoRest.Shared.Tests/AutoRest.Shared.Tests.csproj]: I believe this change comes from merging. Let me revert it.

**Review by m-nash — APPROVED** [review]

**Pull request merged**